src/Engine/ProtoCore/Lang/FunctionGroup.cs

```diff
@@ -187,30 +190,29 @@ public void CopyPublic(List<FunctionEndPoint> rhs)
             //@TODO(Luke): Need to add type statistics checks to the below if it is an array to stop int[] matching char[]
 
             //Now test the reduced Params over all of the available end points
-            // TODO_MSIL: figure out isInstance vs isGlobal
-            //StackValue thisptr = stackFrame.ThisPtr;
-            //bool isInstance = thisptr.IsPointer && thisptr.Pointer != Constants.kInvalidIndex;
-            //bool isGlobal = thisptr.IsPointer && thisptr.Pointer == Constants.kInvalidIndex;
+            bool isGlobal = thisPtr.IsGlobal;
+            bool isInstance = thisPtr.IsPointer && !thisPtr.IsGlobal;
 
             foreach (CLRFunctionEndPoint fep in feps)
             {
-                //var proc = fep.procedureNode;
+                var proc = fep.procedureNode;
 
                 // Member functions are overloaded with thisptr as the first
                 // parameter, so if member function replicates on the left hand
                 // side, the type matching should only be applied to overloaded
                 // member functions, otherwise should only be applied to original
                 // member functions.
 
-                // TODO_MSIL: figure out isInstance vs isGlobal
-                /*if (isInstance && context.IsReplicating != proc.IsAutoGeneratedThisProc)
+                // There are no scenarios yet where isReplicating is true in the MSIL VM.
+                bool isReplicating = false;
```

Comment:
hmm, but shouldn't there be? should this still be a MSIL_TODO? We always replicate?